frontend/micro-ui/web/micro-ui-internals/packages/modules/AttendenceMgmt/src/pageComponents/AttendanceDateRange.js

```diff
@@ -0,0 +1,186 @@
+      const startDate = selectionRange?.startDate;
+      const endDate = selectionRange?.endDate;
+      const duration = getDuration(selectionRange?.startDate, selectionRange?.endDate);
+      const title = `${format(selectionRange?.startDate, "MMM d, yy")} - ${format(selectionRange?.endDate, "MMM d, yy")}`;
+      onFilterChange({ range: { startDate, endDate, duration, title }, requestDate: { startDate, endDate, duration, title } });
+    }
+  }, [selectionRange, isModalOpen]);
+
+  const staticRanges = useMemo(() => {
+    return createStaticRanges([
+      {
+        label: t("DSS_TODAY"),
+        range: () => ({
+          startDate: startOfToday(new Date()),
+          endDate: endOfToday(new Date()),
+        }),
+      },
+      {
+        label: t("DSS_YESTERDAY"),
+        range: () => ({
+          startDate: startOfYesterday(new Date()),
+          endDate: endOfYesterday(new Date()),
+        }),
+      },
+      {
+        label: t("DSS_THIS_WEEK"),
+        range: () => ({
+          startDate: startOfWeek(new Date()),
+          endDate: endOfWeek(new Date()),
+        }),
+      },
+      {
+        label: t("DSS_THIS_MONTH"),
+        range: () => ({
+          startDate: startOfMonth(new Date()),
+          endDate: endOfMonth(new Date()),
+        }),
+      },
+      {
+        label: t("DSS_THIS_QUARTER"),
+        range: () => ({
+          startDate: startOfQuarter(new Date()),
+          endDate: endOfQuarter(new Date()),
+        }),
+      },
+      {
+        label: t("DSS_PREVIOUS_YEAR"),
+        range: () => ({
+          startDate: subYears(addMonths(startOfYear(new Date()), 3), 1),
+          endDate: subYears(addMonths(endOfYear(new Date()), 3), 1),
+        }),
+      },
+      {
+        label: t("DSS_THIS_YEAR"),
+        range: () => ({
+          startDate: addMonths(startOfYear(new Date()), 3),
+          endDate: addMonths(endOfYear(new Date()), 3),
+        }),
+      },
+    ]);
+  }, []);
+
+  const getDuration = (startDate, endDate) => {
+    let noOfDays = (new Date(endDate).getTime() - new Date(startDate).getTime()) / (1000 * 3600 * 24);
+    if (noOfDays > 91) {
+      return "month";
+    }
+    if (noOfDays < 90 && noOfDays >= 14) {
+      return "week";
+    }
+    if (noOfDays <= 14) {
+      return "day";
+    }
+  };
+
+  const handleSelect = (ranges) => {
+    const { range1: selection } = ranges;
+    const { startDate, endDate, title, duration } = selection;
+    if (isStartDateFocused(focusedRange[1])) {
+      setSelectionRange(selection);
+    }
+    if (isEndDateFocused(focusedRange[1])) {
+      setSelectionRange({ title, duration, startDate, endDate: addSeconds(addMinutes(addHours(endDate, 23), 59), 59) });
+      setIsModalOpen(false);
+    }
+  };
+
+  const handleFocusChange = (focusedRange) => {
+    const [rangeIndex, rangeStep] = focusedRange;
+    setFocusedRange(focusedRange);
+  };
+
+  const handleClose = () => {
+    setIsModalOpen(false);
+  };
+
```

Comment:
You can simply import this component from react-comp library if no customization is required @hariomsinha-eGov

Reply:
As discussed, the new file was added due to some underlying changes required based on the UX.